### DISCUSSION TITLE
log messages and conditional break points?

### DISCUSSION THREAD

**Discussion by chriztheanvill** [mfussenegger/nvim-dap#537] [Q&A]
Hi.
I've been using VS Code and switching to neovim it was a hole new adventure. Well, in VS Code, adding a log message you can pass some variables to the message to print the info: `--- someRandomText { mVariableName[key] }` and it will print: `--- someR... { currentValue } `
But here in Dap + Neovim, if I do this with:  `nnoremap <silent> <leader>lp :lua require'dap'.set_breakpoint(nil, nil, vim.fn.input('Log point message: '))<CR>` then run dap, in the "Dap Stacks" just shows what I did and does not show the info of "mVariableName[key]".
Also in the dap-repl does not show.

**Comment by mfussenegger** [comment]
It should output the log messages to the stdout (which may be in the terminal if using the integrated or external terminal, otherwise it should be passed to the repl).

Variable expansion via `{variableName}` should work as well. All of that is done by the debug adapter. The client only informs the debug adapter about those log points.

I use it all the time and it definitely works with adapters who support it properly.

**Comment by chriztheanvill** [comment]
![image](https://user-images.githubusercontent.com/37992958/166000548-5cbe2e6a-6856-46db-900b-1a18bf2164ed.png)
This is what it shows.

**Comment by chriztheanvill** [comment]
Instead of adding some text, just add it: `{mKeyStates[key] }` and shows this:
`<Failed to interpolate {mKeyStates[key]}: -var-create: unable to create variable object `

Thats weird because in VS Code it works:
![image](https://user-images.githubusercontent.com/37992958/166001989-35a018be-9b42-462b-9e66-3de566bdd433.png)

-------
![image](https://user-images.githubusercontent.com/37992958/166002152-e4ada6fe-d193-4101-abdc-0f2dffcfed21.png)

**Comment by chriztheanvill** [comment]
This is my config:
```
,{
			"type": "cppdbg",
			"request": "launch",
			"name": "sdl_yt_mad_03",
			"program": "${workspaceFolder}/build/Debug/bin/sdl_yt_mad_03",
			"args": [],
			"stopAtEntry": false,
			"cwd": "${workspaceFolder}",
			"environment": [],
			"externalConsole": false,
			"MIMode": "gdb",
			"setupCommands": [
				{
					"description": "Enable pretty-printing for gdb",
					"text": "-enable-pretty-printing",
					"ignoreFailures": true
				}
			]
		},
```
